torch/_guards.py

```diff
@@ -409,6 +394,7 @@ def __init__(self, fake_mode):
         self.fake_mode = fake_mode
         self.frame_summary_stack = []
         self.module_context = ModuleContext()
+        self.aot_autograd_arg_pos_to_source = []
```

Comment:
I think my preference is, instead of storing it in the guard state, put it in AOTConfig instead to the use site.

Reply:
This is just general tracing state, but that's fine too. We can also just plumb it through AOT without state...